### PR TITLE
dev:

### PR DESCRIPTION
**Pull request by rupaliT123**
lint:0
coverage:86%
userCreation:
lint:0
coverage:86%

### CODE REVIEW

File: src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem [hunk 1]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (VaibhavTechsophy) [inline src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem:1]: this should not be committed to source code

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalUserFormDataControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import com.techsophy.tsf.account.utils.UserDetails;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.bind.annotation.RestController;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.SAVE_FORM_SUCCESS;
+
+@RestController
+@AllArgsConstructor(onConstructor_ = {@Autowired})
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalUserFormDataControllerImpl.java:20]: Please dont use Allargs constructor

File: src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem (hunk 1 above)
Comment (VaibhavTechsophy) [reply src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem:1]: is this private key?
Private key should not be used in account app module

File: src/main/java/com/techsophy/tsf/account/dto/InternalUserFormDataSchema.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.EqualsAndHashCode;
+
+import javax.validation.constraints.NotEmpty;
+import java.util.Map;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.USER_DATA_NOT_EMPTY;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@AllArgsConstructor
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/dto/InternalUserFormDataSchema.java:14]: please dont ues allargsconstructor

File: src/main/java/com/techsophy/tsf/account/service/impl/InternalUserFormDataServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.List;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.COMMA;
+import static com.techsophy.tsf.account.constants.AccountConstants.USER_DATA_NAME;
+import static com.techsophy.tsf.account.constants.ErrorConstants.UN_AUTHORIZED_EXCEPTION;
+
+@Slf4j
+@Service
+@AllArgsConstructor(onConstructor_ = {@Autowired})
+public class InternalUserFormDataServiceImpl implements InternalUserFormDataService
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/impl/InternalUserFormDataServiceImpl.java:38]: no need for a separate service, please use current implementation

File: src/main/java/com/techsophy/tsf/account/service/InternalUserFormDataService.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.techsophy.tsf.account.service;
+
+import com.techsophy.tsf.account.dto.InternalUserFormDataSchema;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface InternalUserFormDataService
+{
+    @Transactional(rollbackFor = Exception.class)
+    InternalUserFormDataSchema saveUserFormData(String signature, InternalUserFormDataSchema userFormDataSchema);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/InternalUserFormDataService.java:9]: A separate service should not be created, please discuss if there is a solid reason.

File: src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.techsophy.tsf.account.dto.InternalUserFormDataSchema;
+import com.techsophy.tsf.account.model.ApiResponse;
+import org.springframework.http.HttpHeaders;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.VERSION_V1;
+
+@RequestMapping("/internal"+VERSION_V1+"/users/create/")
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java:12]: Lets not create a controller for each internal API, lets keep this as a proxy only

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Have given some comments, please work on those, discuss with me one to one if anything is not clear.
Will further review post their closure

File: src/main/java/com/techsophy/tsf/account/service/InternalUserFormDataService.java (hunk 5 above)
Comment (rupaliT123) [reply src/main/java/com/techsophy/tsf/account/service/InternalUserFormDataService.java:9]: we have created it because it has different functionality we will get the user information and passing them to save with signature verification

File: src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java (hunk 6 above)
Comment (rupaliT123) [reply src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java:12]: only one internal api is there for now

File: src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem (hunk 1 above)
Comment (rupaliT123) [reply src/main/resources/keys/private_key_rsa_4096_pkcs8-generated.pem:1]: we will discuss with @partha-techsophy  , after that i will let you know

File: src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java (hunk 6 above)
Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/account/controller/InternalUserFormDataController.java]: does not matter we need to keep the correct structure

**Comment by rupaliT123** [comment]
lint:0
coverage:85%
userCreation:
lint:0
coverage:85%

File: src/main/java/com/techsophy/tsf/account/controller/InternalController.java [hunk 7]
```diff
@@ -7,10 +7,10 @@
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
 
-import static com.techsophy.tsf.account.constants.AccountConstants.VERSION_V1;
+import static com.techsophy.tsf.account.constants.AccountConstants.*;
 
-@RequestMapping("/internal"+VERSION_V1+"/users/create/")
-public interface InternalUserFormDataController {
+@RequestMapping(INTERNAL+VERSION_V1+USER_CREATE)
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/InternalController.java:12]: dont think I ask this change.

File: src/main/java/com/techsophy/tsf/account/service/InternalService.java [hunk 8]
```diff
@@ -3,7 +3,7 @@
 import com.techsophy.tsf.account.dto.InternalUserFormDataSchema;
 import org.springframework.transaction.annotation.Transactional;
 
-public interface InternalUserFormDataService
+public interface InternalService
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/InternalService.java:6]: please dont waste my time by committing unnecessary changes, if you are not clear on the comments, pls discuss with me directly.

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java [hunk 9]
```diff
@@ -4,28 +4,52 @@
...
 import org.springframework.http.HttpHeaders;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.validation.ConstraintViolationException;
+
 import static com.techsophy.tsf.account.constants.AccountConstants.SAVE_FORM_SUCCESS;
+import static com.techsophy.tsf.account.constants.ErrorConstants.UN_AUTHORIZED_EXCEPTION;
 
 @RestController
 @AllArgsConstructor(onConstructor_ = {@Autowired})
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:25]: please dont use all args constructor

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java [hunk 10]
```diff
@@ -4,28 +4,52 @@
...
-    private final UserDetails userDetails;
     private final ObjectMapper objectMapper;
     @Override
-    public ApiResponse<InternalUserFormDataSchema> saveUser(String signature, InternalUserFormDataSchema userFormDataSchema, HttpHeaders headers) throws JsonProcessingException {
-        return new ApiResponse<>(internalUserFormDataService.saveUserFormData(signature, userFormDataSchema),true,globalMessageSource.get(SAVE_FORM_SUCCESS));
+    public ApiResponse<UserFormDataSchema> saveUser(String signature, UserFormDataSchema internalUserFormDataSchema, HttpHeaders headers) throws JsonProcessingException
+    {
+        try
+        {
+            Rsa4096 rsa4096 = new Rsa4096();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:36]: move this code to a separate reusable method, such that it can be provided a key and it works

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java [hunk 11]
```diff
@@ -4,28 +4,52 @@
...
+            String sign = rsa4096.decryptFromBase64((String) internalUserFormDataSchema.getUserData().get("signature"));
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:37]: why signature is coming from user data?

File: src/main/java/com/techsophy/tsf/account/service/impl/UserFormDataServiceImpl.java [hunk 12]
```diff
@@ -58,14 +56,13 @@ public UserFormDataSchema saveUserFormData(UserFormDataSchema userFormDataSchema
             userDetails.userNameValidations(userData.getUserName());
             String userId = userFormDataSchema.getUserId();
             userData.setUserName(userData.getUserName().toLowerCase());
-            Map<String,Object> loggedInUser = userServiceImpl.getCurrentlyLoggedInUserId().get(0);
-            if (userId == null)
+            BigInteger loggedInUserId = userDetails.getCurrentAuditor().orElse(null);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/account/service/impl/UserFormDataServiceImpl.java:59]: there is no need for orElse(null)

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
I am not really clear on the approach, is the key issue resolved?

**Comment by VaibhavTechsophy** [comment]
@rupaliT123  
It is showing vulnerability and security issues can u please check and evaluate?

File: src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java (hunk 11 above)
Comment (rupaliT123) [reply src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java:37]: When passing Base64 in URL it is converting values such as blank space, and other special characters according to url pattern, and decode is faling, therefore we are passing it in body

**Comment by VaibhavTechsophy** [comment]
Closing this merge request for now, once all inputs are with you raise a fresh one.
